query-node/joystream-query-node/schema.graphql

```diff
@@ -1,21 +1,291 @@
+	"Name of property"
+	name: String!
+
+	"Description of property"
+	description: String!
+
+	"Property is locked from maintainer"
+	isLockedFromMaintainer: Boolean!
+
+	"Property is locked from controller"
+	isLockedFromController: Boolean!
+
+	"Upper bound for vector length when property is a vector"
+	maximumLength: BigInt
+
+	"Optional field, it must reference a class then PropertyType is Reference"
+	referencedClass: Class
+
+	"Optional field, Only will be set when PropertyType is Reference"
+	sameControllerStatus: Boolean
+
+	"Schemas which involve this property"
+	schemas: [Schema!] @derivedFrom(field: "properties")
+}
+
+"Represents a standard for what information must be associated with any entity of a given class"
+type Schema @entity {
+	"Runtime identifier of schema in class"
+	schemaId: BigInt!
+
+	"Class to which this schema corresponds"
+	class: Class!
+
+	"Properties of class that are in this schema"
+	properties: [Property!]
+
+	"If schema can be added to an entity"
+	isActive: Boolean!
+}
+
+"Working group participant: working-group/src/types.rs"
+type Worker @entity {
+	member: Member!
+
+	"Account used to authenticate in this role"
+	roleAccountId: Bytes!
+
+	# "Whether the role has recurring reward, and if so an identifier for this"
+	# rewardRelationship: Option<RewardRelationshipId>
+
+	# "When set, describes role stake of the worker/lead"
+	# roleStakeProfile: Option<RoleStakeProfile<StakeId, BlockNumber>>
+}
+
+type Curator @entity {
+	"Account used to authenticate"
+	worker: Worker!
+
+	groups: [CuratorGroup] @derivedFrom(field: "curators")
+}
+
+"A group, that consists of 'curators' set"
+type CuratorGroup @entity {
+	"Runtime identifier for group"
+	groupId: BigInt!
+
+	"Activity stats of group"
+	isActive: Boolean!
+
+	"All member curators of this group"
+	curators: [Curator!]
+
+	"All classes this group maintains"
+	maintainerOfClasses: [Class!] @derivedFrom(field: "maintainers")
+}
+
+"A voucher for 'Entity' creation"
+type EntityCreationVoucher @entity {
+	"Class to which voucher applies"
+	class: Class!
+
+	"Controller to which this voucher applies"
+	controller: EntityController!
+
+	"How many are allowed in total"
+	maximumEntitiesCount: BigInt!
+
+	"How many have currently been created"
+	entitiesCreated: BigInt!
+}
+
+enum EntityControllerType {
+	LEAD
+	MEMBER
+	MAINTAINERS
+}
+
+"Owner of an 'Entity'"
+type EntityController @entity {
+	"Current controller, which is initially set based on who created entity"
+	controllerType: EntityControllerType!
+
+	"Optional field, will be set when EntityControllerType is MEMBER"
+	controllerWhenMember: Member
+}
+
+"""
+Represents a particular instance of a class,
+hence each entity corresponds to exactly one class
+"""
+type Entity @entity {
+	"Runtime identifier for entity"
+	entityID: BigInt!
+
+	"Current controller, which is initially set based on who created entity"
+	controller: EntityController!
+
+	"Forbid groups to mutate any property value"
+	isFrozen: Boolean!
```

Comment:
```
"Current controller, which is initially set based on who created entity"
	controller: EntityController!
	"Forbid groups to mutate any property value"
	isFrozen: Boolean!
"Prevent from being referenced by any entity (including self-references)"
	isReferenceable: Boolean!
```
Can we move `EntityPermissions` related fields to separate type, please?

Reply:
If we have `EntityPermissions` type there is no use except in `Entity` so I think it's good to keep the fields here.

Reply:
I don`t think someone will query only entity permissions, though it seems to be easier to select all permissions related fields, when he/she need them. 

```
query {
    class {
      entityId
      entityPermissions
  }
}
```

instead of 

```
query {
    class {
      entityId,
     controller: EntityController!
	"Forbid groups to mutate any property value"
	isFrozen: Boolean!
"Prevent from being referenced by any entity (including self-references)"
	isReferenceable: Boolean!
  }
}
```

Will it complicate things a lot?